modular_skyrat/modules/new_meteors/code/new_meteors.dm

```diff
@@ -0,0 +1,78 @@
+/// Each tick lasts two seconds.
+#define MINUTES_IN_TICKS 30
+#define METEOR_WAVE_MIN_NOTICE 7 * MINUTES_IN_TICKS
+#define METEOR_WAVE_MAX_NOTICE 9 * MINUTES_IN_TICKS
+#define METEOR_WAVE_DURATION 1.25 * MINUTES_IN_TICKS
+#define METEOR_TICKS_BETWEEN_WAVES 3
+#define METEOR_PER_WAVE 5
+
+/datum/round_event_control/meteor_wave
+	///Where will the meteors be coming from? -- Established in admin_setup, passed down to round_event
+	var/start_side
+
+/datum/round_event/meteor_wave
+	///Which direction the storm will come from.
+	var/start_side
+
+/datum/round_event/meteor_wave/New()
+	. = ..()
+	start_when = rand(METEOR_WAVE_MIN_NOTICE, METEOR_WAVE_MAX_NOTICE)
+	end_when = start_when + METEOR_WAVE_DURATION
+
+/datum/round_event_control/meteor_wave/admin_setup()
+	if(!check_rights(R_FUN))
+		return ADMIN_CANCEL_EVENT
+
+	var/force_dir = tgui_alert(usr, "Choose a side to throw rocks at?", "Interrupt some ERP?", list("Yes", "No", "Cancel"))
+	if(force_dir == "Cancel")
+		return ADMIN_CANCEL_EVENT
+
+	if(force_dir == "Yes")
+		var/chosen_direction = tgui_input_list(usr, "Pick one!","Plausible Deniability.", list("North", "South", "East", "West", "Cancel"))
+		switch(chosen_direction)
+			if("North")
+				start_side = NORTH
+			if("South")
+				start_side = SOUTH
+			if("East")
+				start_side = EAST
+			if("West")
+				start_side = WEST
+			else
+				return ADMIN_CANCEL_EVENT
+
+	if(force_dir != "Yes")
+		return
+
+/datum/round_event/meteor_wave/announce(fake)
+	var/datum/round_event_control/meteor_wave/meteor_event = control
+	start_side = meteor_event.start_side || pick(GLOB.cardinals)
+
+	var/start_side_text = "unknown"
```

Comment:
Did we ever get an update on the PR to make a proc for nautical directions?

Reply:
Yeah it died at TG. Maybe after freeze.